### PR TITLE
[SDESK-2777] Only support users can view support users

### PR DESCRIPTION
**Pull request by pablopunk**
Following https://github.com/superdesk/superdesk-core/pull/1252

### CODE REVIEW

File: scripts/apps/users/directives/UserListDirective.js [hunk 1]
```diff
@@ -25,6 +25,15 @@ export function UserListDirective(keyboardManager, usersService, asset) {
                 bindKeys();
             };
 
+            scope.$watch('users', (users) => {
```
Comment (petrjasek) [inline scripts/apps/users/directives/UserListDirective.js]: I would rather put this logic into `UserListController` where is the rest of the filtering (active/online/etc) so it's all in one place

**Comment by pablopunk** [comment]
I found an error, don't merge this

File: scripts/apps/users/controllers/UserListController.js [hunk 2]
```diff
@@ -117,6 +117,15 @@ export function UserListController($scope, $location, api, _) {
             break;
         }
 
+        if (!usersService.isSupport(session.identity)) {
+            // If user is not support type, they can't see support users
+            query.$or = [
```
Comment (petrjasek) [inline scripts/apps/users/controllers/UserListController.js]: I think it should be like `query.is_support: {$ne: true}`

File: scripts/apps/users/controllers/UserListController.js [hunk 3]
```diff
@@ -75,14 +75,28 @@ export function UserListController($scope, $location, api, _, session, usersServ
...
 
         if (search.q) {
             query = {
                 $or: [
-                    {username: {$regex: search.q, $options: '-i'}},
-                    {display_name: {$regex: search.q, $options: '-i'}},
-                    {email: {$regex: search.q, $options: '-i'}},
+                    {
+                        username: {$regex: search.q, $options: '-i'},
+                        is_support: {$ne: !canSeeSupportUsers},
```
Comment (petrjasek) [inline scripts/apps/users/controllers/UserListController.js]: I think you don't have to add it in `$or`, the query should work in a way that on top level it's AND so it must pass is_support + some of the $or filters

**Pull request merged**